docs(python): rewrite page (#16465)
renovatebot/renovate: docs/usage/python.md

## Versioning support

> We've written a TypeScript version of the [PEP440 specification](https://www.python.org/dev/peps/pep-0440/) so we can use it in Renovate bot.

Author: Check me if changing the programming language to TypeScript is correct.

Reviewer: No, it's actually in JS

Author: ```suggestion
We've written a JavaScript version of the [PEP440 specification](https://www.python.org/dev/peps/pep-0440/) so we can use it in Renovate bot.
```

You can find this project in our [`renovatebot/pep440` repository](https://github.com/renovatebot/pep440).

Our PEP440 implementation supports pinned versions and ranges.
Legacy versions with the `===` prefix are ignored.

## How it works

1. Renovate searches through each repository for package files
1. Existing dependencies are extracted from the package files
1. Renovate searches for the latest version on [PyPI](https://pypi.org/) to decide if there are upgrades
1. If the source package includes a GitHub URL as its source, and has a "changelog" file _or_ uses GitHub releases, a Release Note will be embedded in the generated PR

## Alternative file names

> The default file matching regex for `requirements.txt` follows common file name conventions.

Reviewer 2: this is only true for our `pip_requirements` manager. we should probably move this section to the manager readme

### Sources in `Pipfile`

> Renovate finds and uses any custom sources in your `Pipfile`.

Author: I think we should add an example on how to set a custom source in your `Pipfile`.

<!-- prettier-ignore -->
!!! tip
    If a `requirements.txt` file has a index-url then Renovate follows that link, instead of following any link set in the `registryUrls` array.

>     To override the URL found in `requirements.txt`, you must create a custom `packageRules` setting.

Author: I think this tip should get an example as well. After reading this text you still don't know _what_ you should put in the `packageRules`.

Reviewer 2: add a link to private packages page and add samples there

Author: Which page do you mean, can you give me a link?

And can you give me the example to put in the private packages page?

Reviewer: https://docs.renovatebot.com/getting-started/private-packages/

    This is because `packageRules` are applied _after_ package file extraction.

## Disabling Python support

> To disable _all_ Python support in Renovate, do this:

Reviewer 2: not sure if this is still true 🤔

<!-- prettier-ignore -->
!!! tip
    If a `requirements.txt` file has a index-url then Renovate follows that link, instead of following any link set in the `registryUrls` array.
    To override the URL found in `requirements.txt`, you must create a custom `packageRules` setting.
    This is because `packageRules` are applied _after_ package file extraction.
    OPTION 1: Create new secion in private packages docs, and add this link: Go to the [private package support docs](https://docs.renovatebot.com/getting-started/private-packages/#pip).

>     OPTION 2: Link to https://docs.renovatebot.com/python/#specify-url-in-configuration

Author: I found a section on the "Python package manager support" page [^specify-url-in-configuration] which explains how you can set a `registryUrls` array:

> ### Specify URL in configuration
>
> You can use the `registryUrls` array to configure alternate index URL(s). e.g.:
>
> ```json
> {
>  "python": {
>    "registryUrls": ["http://example.com/private-pypi/"]
>  }
> }
> ```

But the `registryUrl` config option docs [^registry-url-config-option] has this example:

> ```json
> {
>   "packageRules": [
>     {
>       "matchDatasources": ["docker"],
>       "registryUrls": ["https://docker.mycompany.domain"]
>     }
>   ]
> }
> ```

One example uses a `"python"` object, the other uses a `matchDatasources` config option. I don't know which one is correct when you have you want to override the `index-url` entry in your `requirements.txt` file... 😄 

[^specify-url-in-configuration]: https://docs.renovatebot.com/python/#specify-url-in-configuration
[^registry-url-config-option]: https://docs.renovatebot.com/configuration-options/#registryurls

>     OPTION 2: Link to https://docs.renovatebot.com/python/#specify-url-in-configuration

Author: ```suggestion
    OPTION 2: Link to [specify URL in Python docs](https://docs.renovatebot.com/python/#specify-url-in-configuration)
```

## Disabling Python support

> To disable _all_ Python support in Renovate, do this:

Reviewer: ```suggestion
To disable all managers with `language` set to `python` in Renovate, do this:
```

}
```

> Alternatively, you can use `enabledManagers` to tell Renovate what package managers it can use.

Author: I'm not sure we need a section on enabling specific managers, instead of disabling Python globally. It feels a bit out of place.

Reviewer: Agreed, much of this document can probably be removed

Reviewer: I want to remove this page later anyway, no sense to do too much work now
